SmartcarAuth/SmartcarAuth.swift

```diff
@@ -144,19 +143,32 @@ Smartcar Authentication SDK for iOS written in Swift 3.
         let urlComp = URLComponents(url: url, resolvingAgainstBaseURL: false)
 
         guard let query = urlComp?.queryItems else {
-            return completion(AuthorizationError.missingQueryParameters, nil, nil)
+            return completion(AuthorizationError.missingQueryParameters, nil, nil, nil)
         }
 
         let queryState = query.filter({ $0.name == "state"}).first?.value
-
-        if query.filter({ $0.name == "error"}).first?.value != nil {
-            return completion(AuthorizationError.accessDenied, nil, queryState)
+        var vehicle = Vehicle()
+        vehicle.vin = query.filter({ $0.name == "vin"}).first?.value ?? ""
+        vehicle.make = query.filter({ $0.name == "make"}).first?.value ?? ""
+        vehicle.model = query.filter({ $0.name == "model"}).first?.value ?? ""
+        vehicle.year = query.filter({ $0.name == "year"}).first?.value ?? ""
+        let error = query.filter({ $0.name == "error"}).first?.value
+        
+        if (error != nil) {
+            switch (error) {
```

Comment:
Nit - it's a little weird that we have a guard right below this, but an if check for nil here. I would prefer the guard, tbh

```
guard let errorString = error else {
    // switch here
}
```

Reply:
My reasoning for this is because guards are meant for function stopping errors like this. So when parsing a file, upon reading `guard`, I know that if that condition fails it is a fatal error.

Reply:
Rishab resolved this offline